proxy/proxy.go

```diff
@@ -68,27 +69,31 @@ func setupLoggerBackend(level logging.Level, writer io.Writer) logging.LeveledBa
 	return leveler
 }
 
+// Currency :  Handles logging and RPC details
 type Currency struct {
 	log        *logging.Logger
 	jsonHandle codec.JsonHandle
 
 	params map[string]string
 
-	ticker  string
-	rpcUser string
-	rpcPass string
-	rpcUrl  string
+	ticker   string
+	chaindID int
+	rpcUser  string
+	rpcPass  string
+	rpcURL   string
 }
 
+// Parameters : Returns params from Currency struct
```

Comment:
With getters, we can omit passing the error along.

Reply:
Meaning, this method would be something like:

```go
func (k *Currency) Parameters() (map[string]string) {
   return k.params
}
```

Reply:
Your plugin must implement the ServicePlugin interface:
https://github.com/katzenpost/server/blob/master/cborplugin/client.go#L63-L79

Reply:
It's not clear to me how to implement the `Halt()` function. In its current form, the currency plugin doesn't need it. A refactoring should be done in order to make use of `Halt` for graceful shutdown of the plugin.

Reply:
I think it's clear that the Halt method would be a no-op in your case since you do not need to actually stop any goroutines or free any resources. Anyway, the Halt method must exist otherwise your plugin will not be implementing the ServicePlugin interface.